setwd("/pub/jenyuw/EE283/ATACseq/")

## load the library
library("tidyverse")
library("BiocManager")
library("ATACseqQC")
library(Rsamtools)
library(GenomicFeatures)
library(ChIPpeakAnno)
library(BSgenome.Dmelanogaster.UCSC.dm6)
library(TxDb.Dmelanogaster.UCSC.dm6.ensGene)



##USE the gtf from flybase
library(TxDb.Dmelanogaster.UCSC.dm6.ensGene)

# load a bamfile in the working directory
bamfile <- "/pub/jenyuw/EE283/ATACseq/results/processed_bam/A4.ED.2.dedup.bam"
# BAM file name
bamfile.labels <- gsub(".dedup.bam", "", basename(bamfile))
# generate fragement size distribution
fragSize <- fragSizeDist(bamfile, bamfile.labels)
estimateLibComplexity(readsDupFreq(bamfile, index=bamfile))

tags <- c("AS", "XN", "XM", "XO", "XG", "NM", "MD", "YS", "YT")

possibleTag <- combn(LETTERS, 2)
possibleTag <- c(paste0(possibleTag[1, ], possibleTag[2, ]),
                 paste0(possibleTag[2, ], possibleTag[1, ]))

bamTop100 <- scanBam((bamfile), yieldSize = 100, param = ScanBamParam(tag = possibleTag))[[1]]$tag
tags <- names(bamTop100)[lengths(bamTop100)>0]

outPath <- "splited"
dir.create(outPath)

library(BSgenome.Dmelanogaster.UCSC.dm6)
gal <- readBamFile(bamfile, tag=tags, asMates=TRUE)
gal1 <- shiftGAlignmentsList(gal)
shiftedBamfile <- file.path(outPath, "shifted.bam")
export(gal1, shiftedBamfile)



##USE the gtf from flybase
library(TxDb.Dmelanogaster.UCSC.dm6.ensGene)
txs = transcripts(TxDb.Dmelanogaster.UCSC.dm6.ensGene)
txs = gsub("chr","",txs) ##Because we aligned the reads to Flybase's genome
TSS = promoters(txs, upstream=0, downstream=1)
TSS = unique(TSS)
TSS = gsub("chr","",TSS) ##Because we aligned the reads to Flybase's genome
# At this point TSS is no longer GFrames, you need to coerce it back into GFrames; e.g. 
TSS = GFrames(TSS)

objs = splitGAlignmentsByCut(gal1, txs = txs, outPath = outPath)
null = writeListOfGAlignments(objs, outPath)
## list the files generated by splitBam.
dir(outPath)

bamfiles = file.path(outPath, c("NucleosomeFree.bam","mononucleosome.bam","dinucleosome.bam","trinucleosome.bam"))

# estimate the library size for normalization
library(ChIPpeakAnno)
librarySize = estLibSize(bamfiles)
# Only consider these chromosomes (pick one for speed/memory)
seqlev = "chr2L"
NTILE = 101
dws = ups = 1010


seqlev = "X"
sigs = enrichedFragments(bamfiles, TSS=TSS, librarySize=librarySize, seqlev=seqlev, TSS.filter=0.5, n.tile = NTILE, upstream = ups, downstream = dws)

names(sigs) <- gsub(".bam", "", basename(names(sigs)))
sigs.log2 <- lapply(sigs, function(.ele) log2(.ele+1))

featureAlignedHeatmap(sigs.log2, reCenterPeaks(TSS, width=ups+dws), zeroAt=.5, n.tile=NTILE)


library(GenomicFeatures)
txdbmm = makeTxDbFromGFF(file = "/pub/jenyuw/EE283/Reference/dmel-all-r6.13.gtf")
txs2 = transcripts(txdbmm)
TSS2 = promoters(txs2, upstream=0, downstream=1)
TSS2 = unique(TSS) 